server/src/internalClusterTest/java/org/opensearch/indices/IndicesRequestCacheIT.java

```diff
@@ -677,14 +689,324 @@ public void testCacheWithInvalidation() throws Exception {
+
+        // force refresh so that it creates 2 stale keys in the cache for the cache cleaner to pick up.
+        flushAndRefresh("index");
+        client().prepareIndex("index").setId("1").setSource("k", "good bye");
+        ensureSearchable("index");
+
+        // create another entry
+        createCacheEntry(client, index, "hello1");
+        assertCacheState(client, "index", 0, 3);
+        long currentMemorySize = getRequestCacheStats(client, "index").getMemorySizeInBytes();
+        long expectedThirdCachedEntrySize = currentMemorySize - (expectedFirstCachedItemEntrySize * 2);
+        assertTrue(currentMemorySize > expectedFirstCachedItemEntrySize * 2);
+        assertEquals(currentMemorySize, expectedFirstCachedItemEntrySize * 3, 2);
+
+        Instant end = Instant.now();
+        long elapsedTimeMillis = Duration.between(start, end).toMillis();
+        // if this test is flaky, increase the sleep time.
+        long sleepTime = (thresholdInMillis - elapsedTimeMillis) + 2_000;
+        Thread.sleep(sleepTime);
+
+        // cache cleaner should have cleared the stale keys by now
+        currentMemorySize = getRequestCacheStats(client, "index").getMemorySizeInBytes();
+        // cache should only contain the third entry, the first 2 entries should have been cleaned up
+        assertEquals(currentMemorySize, expectedThirdCachedEntrySize);
+    }
+
+    // when staleness threshold is high, it should NOT clean-up
+    public void testStaleKeysCleanup_HighThresholdSkipsCleanUp() throws Exception {
+        Instant start = Instant.now();
+        long thresholdInMillis = 1_500;
+        String node = internalCluster().startNode(
+            Settings.builder()
+                .put(IndicesRequestCache.SETTING_INDICES_REQUEST_CACHE_STALENESS_THRESHOLD_SETTING, 0.90)
+                .put(IndicesRequestCache.SETTING_INDICES_REQUEST_CACHE_CLEAN_INTERVAL_SETTING, TimeValue.timeValueMillis(thresholdInMillis))
+        );
+        String index = "index";
+        Client client = client(node);
+        setupIndex(client, index);
+
+        // create first cache entry
+        createCacheEntry(client, index, "hello");
+        assertCacheState(client, index, 0, 1);
+        long expectedFirstCachedItemEntrySize = getRequestCacheStats(client, index).getMemorySizeInBytes();
+        assertTrue(expectedFirstCachedItemEntrySize > 0);
+
+        // create second cache entry
+        createCacheEntry(client, index, "there");
+        assertCacheState(client, "index", 0, 2);
+        assertEquals(expectedFirstCachedItemEntrySize * 2, getRequestCacheStats(client, "index").getMemorySizeInBytes());
+
+        // force refresh so that it creates 2 stale keys in the cache for the cache cleaner to pick up.
+        flushAndRefresh("index");
+        client().prepareIndex("index").setId("1").setSource("k", "good bye");
+        ensureSearchable("index");
+
```

Comment:
this will be flaky as well since the moment you triggered `flush/refresh` all your entries will be stale which means 100% is stale and cleanup can happen before adding 3rd entry in cache. 

You can again simulate using above semantics of 2 indices. Create 2 cache entries for each index and keep threshold as 60% and then trigger flush on one of the index and assert after sleeping for cache cleanup interval that stale entries are not cleaned up.